### ISSUE TITLE
Quick Question on API upload and publishing option

### ISSUE THREAD

**Issue by pradeepgorle13** [apigee/apigee-devportal-kickstart-drupal#424]
Hi Guys,

Need your guidance on one thing.
Under /admin/content/api/add 
I want to make sure to divide the API Publisher to API Developer and API Owner.
If you see the below screenshot, the API Developer should be able to upload the API Spec and it will be informed to the API Owner who in turn will publish it.
![image](https://user-images.githubusercontent.com/56166711/93275410-d2a81600-f78a-11ea-8f34-7c49c450769f.png)
In short, I will create two Roles. One Will have the ability to upload the API Spec document and other can upload as well as publish the Document. Do we have any permissions in the that can be separated in the Permissions tab as shown below
![image](https://user-images.githubusercontent.com/56166711/93275664-7f829300-f78b-11ea-9cc3-6e1cd16d3f56.png)
OR do we have a quick and easy option to do that in another way.
Thanks.,

**Comment by arlina-espinoza** [comment]
Hi @pradeepgorle13 ,

It looks like you are using `apigee_api_catalog` 1.x. If you update to the 2.x branch you'll find the API Doc custom entity was migrated into nodes. Once your project is on the 2.x branch with nodes, you can setup an editorial workflow using the core module "Content Moderation" that will allow a role to edit nodes, and another role can have permission to publish.

There should be many articles out there describing how to use the content moderation module, but here are a couple of links to get started:
- [Core docs for the Content Moderation module](https://www.drupal.org/docs/8/core/modules/content-moderation/overview)
- [Drupal 8 Publishing Workflows and Content Moderation](https://atendesigngroup.com/articles/drupal-8-publishing-workflows-and-content-moderation)

**Comment by arlina-espinoza** [comment]
Closing, but feel free to reopen if there are further questions.
